docs/tutorialkit.dev/src/content/docs/guides/using-low-level-apis.mdx

```diff
@@ -0,0 +1,189 @@
+
+```ts
+import tutorialStore from "tutorialkit:store";
+
+console.log(tutorialStore.documents.value);
+{
+  "/index.js": {
+    "value": "export default 'hello from index.js';\n",
+    "type": "file",
+  },
+  "/vite.config.ts": {
+    "value": "import { defineConfig } from \"vite\";\n\nexport default defineConfig({});\n",
+    "type": "file",
+  }
+}
+```
+
+It also exposes methods that can be used to modify lesson files:
+
+```ts
+import tutorialStore from "tutorialkit:store";
+
+await tutorialStore.addFolder("/src");
+await tutorialStore.addFile("/src/new-file.js");
+tutorialStore.updateFile("/src/new-file.js", "console.log('Hello world');");
+```
+
+### Example usage in React component
+
+Here is an example of custom React component that reads contents of `index.js` and gives user hints about the expected content.
+
+<HelpButton client:load />
+
+```tsx title="src/components/HelpButton.tsx"
+import tutorialStore from "tutorialkit:store";
+import { Dialog } from "@tutorialkit/react";
+import { useState } from "react";
+
+export function HelpButton() {
+  const [message, setMessage] = useState<string | null>(null);
+
+  function onClick() {
+    const files = tutorialStore.documents.value;
+    const index = files["/index.js"].value as string;
+
+    if (index.length === 0) {
+      setMessage("Your index.js should not be empty");
+    } else if (!index.includes("export default")) {
+      setMessage("Your index.js should have a default export");
+    } else {
+      setMessage("All good");
+    }
+  }
+
+  return (
+    <>
+      <button
+        onClick={onClick}
+        className="px-4 py-1 rounded-md bg-tk-elements-primaryButton-backgroundColor text-tk-elements-primaryButton-textColor"
+      >
+        Help
+      </button>
+
+      {message && (
+        <Dialog title="Help" confirmText="OK" onClose={() => setMessage(null)}>
+          {message}
+        </Dialog>
+      )}
+    </>
+  );
+}
+```
+
+```mdx title="content.mdx"
+---
+type: lesson
+title: Lesson Topic
+---
+
+import { HelpButton } from "@/components/HelpButton";
+
+# Lesson Topic
+
+In this lesson we'll learn about...
+Click this button if you need help: <HelpButton client:load />
+```
+
+## Tutorial Core
+
+You can access Tutorial Core by importing the `tutorialkit:core` entrypoint.
+
+```ts
+import * as core from "tutorialkit:core";
+```
+
+The Tutorial Core provides access to webcontainer APIs. You can use it for example to read and modify files on the virtual file system.
+
+```ts
+import * as core from "tutorialkit:core";
+
+const webcontainer = await core.webcontainer;
+const contents = await webcontainer.fs.readdir("/", "utf8");
+
+console.log(contents);
+[
+  "index.html",
+  "node_modules",
+  "package-lock.json",
+  "package.json",
+  "vite.config.ts"
+]
+```
+
+### Example usage in React component
+
+Here is an example of custom React component that looks for updates in `/index.js` and provides user information about their lesson status.
```

Comment:
I think I would have this in its own "How to guide" to make it more discoverable. And maybe use a `babel` or `acorn` to be more useful.